elytron/src/main/java/org/wildfly/extension/elytron/CertificateAuthorityAccountDefinition.java

```diff
@@ -122,11 +87,11 @@ public static CertificateAuthority fromName(final String name) {
             .setRestartAllServices()
             .build();
 
-    static final SimpleAttributeDefinition CERTIFICATE_AUTHORITY = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY, ModelType.STRING, true)
+    static final SimpleAttributeDefinition CERTIFICATE_AUTHORITY = new CertificateAuthorityDefinitionBuilder(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY, ModelType.STRING, true)
             .setDefaultValue(new ModelNode(CertificateAuthority.LETS_ENCRYPT.getName()))
```

Comment:
In this area I think we can simplify even further.

I would recommend dropping the setDefaultValue call on the CERTIFICATE_AUTHORITY attribute and instead ensure this attribute can be left undefined.

Instead of checking for the String "LetsEncrypt" checking if the attribute is undefined would be safer, this would also remove the need for the special capability reference handling.

Overall I think your solution does look really nice but I think there could be confusion in the future if an administrator added a definition for a LetsEncrypt certificate authority, this could be a possibility if they ever changed their URLs - the administrator would then try setting this attribute to "LetsEncrypt" and ask why their configuration is not detected.

Reply:
@darranl Thank you for review. If administrator tries to add LetsEncrypt it throws an exception saying it is configured by default. If different URLs are needed, new resource with different name can be added. This solution was done to avoid adding LetsEncrypt to default configuration.

Reply:
Thanks @Skyllarr if they get an error trying to add the CA that does address my concern.

Reply:
@darranl I think that the special capability reference handling must be done for the scenario where administrator sets LetsEncrypt specifically. Eg this should work because of backwards compatibility:

```xml
<certificate-authority-account name="MyCA" certificate-authority="LetsEncrypt">
        ....
</certificate-authority-account>
```

Reply:
Yes as you diaslow them from defining their own 'LetsEncrypt' CA that makes sense, there is not going to be any ambiguity.